db/controllers/controllers.js

```
@@ -46,8 +47,13 @@ exports.getCommentsByArticleId = (req, res, next)=>{
     })
 }
 exports.getAllArticles = (req,res,next)=>{
-    fetchAllArticles()
-    .then((articles)=>{
+   const {topic}=req.query
+
+    const promises=[fetchAllArticles(topic)]
+    if(topic){ promises.push(fetchTopicsBySlug(topic))}
```

At this stage we should only need one model to be queried, what's the reason why we are querying two models here? With the tests that you have written we could rely on the one model and check the response for our error handling. Have a think about what tests we have written so far and what code those tests are driving us to write.

---

~~I did this so that if a topic that doesn't exist is used in a query it will throw a 404 error, rather than just an empty array 200, which I took to mean that it is a valid topic but just with no articles with it as a topic, rather than just not even attempting if the topic isn't in the database...?  Something Paul talked about in his lecture led me to believe this was the correct approach but maybe I misinterpreted...?~~